src/Tools/_framework/Paths/PortfolioActivitySettings.jsx

```diff
@@ -39,7 +39,7 @@ export async function action({ request, params }) {
   });
   const portfolioCourseId = response?.data?.portfolioCourseId;
 
-  if (updates.makePublicPrivate) {
+  if (updates.makePublicPrivate == 'true') {
     let response = await fetch(
```

Comment:
Taking another look at this, It looks like this code might not support marking an activity private again after being made public, I think we need to make a request in either case just with the parameter changed right?